### PR TITLE
Single CI workflow for unified github action report

### PR DESCRIPTION
**Pull request by touilleMan**
Accessing all tests jobs in a single report:
image

This also means we only have to configure `ci-is-happy` as the required job in the github branch protection

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+        run: poetry run pytest ${{ env.pytest-base-args }} tests --runmountpoint --runslow -m mountpoint
+        timeout-minutes: 10
+
+      - name: (🐧 Linux) Install pytest-xvfb plugin for pytest for Qt Testing
+        if: >-
+          steps.python-changes.outputs.run == 'true'
+          && startsWith(matrix.os, 'ubuntu-')
+        run: poetry run pip install pytest-xvfb
+
+      - name: (🐧&🏁 Not macOS) GUI tests
```
Comment (FirelightFlagboy) [inline .github/workflows/ci.yml:215]:
```suggestion
      - name: (🐧 & 🏁 Not macOS) GUI tests
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+          set -x
+          poetry run python --version
+          poetry run pip --version
+
+      - name: Basic tests
+        if: steps.python-changes.outputs.run == 'true'
+        run: poetry run pytest ${{ env.pytest-base-args }} tests -n auto
+        timeout-minutes: 10
+
+      - name: (🐧&🏁 Not macOS) Mountpoint tests
```
Comment (FirelightFlagboy) [inline .github/workflows/ci.yml:202]:
```suggestion
      - name: (🐧 & 🏁 Not macOS) Mountpoint tests
```

**Review by FirelightFlagboy — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (touilleMan) [reply .github/workflows/ci.yml]: I've removed the & entirely then (less clutter this way)

File: .github/workflows/ci.yml (hunk 2 above)
Comment (touilleMan) [reply .github/workflows/ci.yml]: I've removed the & entirely then (less clutter this way)

**Review by FirelightFlagboy — APPROVED** [review]

**Pull request merged**